Can't use Proxy Authentication without password

**Describe the bug**
I'm working in a project with N-tier authentication. The middle tier does not know the password of the proxy user.
Everything works as expected before godror v0.10.3.

**To Reproduce**
Add a new test case to z_heterogeneous_test.go
```go
//line 82
"proxyUser":       {In: godror.ContextWithUserPassw(ctx, proxyUser, proxyPassword, ""), Want: proxyUser},
"proxyUserNoPass": {In: godror.ContextWithUserPassw(ctx, proxyUser, "", ""), Want: proxyUser},
```

**Expected behavior**
Test PASS.

**Error output**

ORA-24419: Proxy sessions are not supported in this mode. 

```
TestHeterogeneousPoolIntegration/proxyUserNoPass: z_heterogeneous_test.go:81: user="test_proxyUserX" ConnectString="localhost:1521/xe" stats=busy=0 open=1 max=16 maxLifetime=5m0s timeout=1m0s waitTimeout=5s params={authMode:0 connectionClass:<nil> connectionClassLength:0 purity:0 newPassword:<nil> newPasswordLength:0 appContext:<nil> numAppContext:0 externalAuth:0 externalHandle:<nil> pool:0x7f91e8046eb0 tag:<nil> tagLength:0 matchAnyTag:0 outTag:<nil> outTagLength:0 outTagFound:0 shardingKeyColumns:<nil> numShardingKeyColumns:0 superShardingKeyColumns:<nil> numSuperShardingKeyColumns:0 outNewSession:0}: ORA-24419: Proxy sessions are not supported in this mode.
--- FAIL: TestHeterogeneousPoolIntegration (0.73s)
    --- PASS: TestHeterogeneousPoolIntegration/noContext (0.08s)
    --- PASS: TestHeterogeneousPoolIntegration/proxyUser (0.10s)
    --- FAIL: TestHeterogeneousPoolIntegration/proxyUserNoPass (0.00s)
```

**Your oracle client version**
19.3.0.0.0

**Your godror version**
v0.20.5

**Your go version**
1.14.3

**Your gcc version**
gcc version 10.2.0 (Debian 10.2.0-15)

**Machine (please complete the following information):**
 - OS: [Debian]
 - Architecture [x86_64]
 - Version: [10.2]

**Additional context**
Add any other context about the problem here.

--- a/drv.go
+++ b/drv.go
@@ -523,13 +523,13 @@ func (d *drv) getPool(P commonAndPoolParams) (*connPool, error) {
 		return nil, err
 	}
 
-	if P.Heterogeneous {
-		P.Username = ""
-		P.Password.Reset()
+	var usernameKey string
+	if !P.Heterogeneous {
+		usernameKey = P.Username
 	}
 	// determine key to use for pool
 	poolKey := fmt.Sprintf("%s\t%s\t%d\t%d\t%d\t%s\t%s\t%s\t%t\t%t\t%t",
-		P.Username, P.ConnectString, P.MinSessions, P.MaxSessions,
+		usernameKey, P.ConnectString, P.MinSessions, P.MaxSessions,
 		P.SessionIncrement, P.WaitTimeout, P.MaxLifeTime, P.SessionTimeout,
 		P.Heterogeneous, P.EnableEvents, P.ExternalAuth)
 	if Log != nil {

--- a/z_heterogeneous_test.go
+++ b/z_heterogeneous_test.go
@@ -70,8 +70,9 @@ func TestHeterogeneousPoolIntegration(t *testing.T) {
 		In   context.Context
 		Want string
 	}{
-		"noContext": {In: ctx, Want: username},
-		"proxyUser": {In: godror.ContextWithUserPassw(ctx, proxyUser, proxyPassword, ""), Want: proxyUser},
+		"noContext":       {In: ctx, Want: username},
+		"proxyUser":       {In: godror.ContextWithUserPassw(ctx, proxyUser, proxyPassword, ""), Want: proxyUser},
+		"proxyUserNoPass": {In: godror.ContextWithUserPassw(ctx, proxyUser, "", ""), Want: proxyUser},
 	} {
 		t.Run(tName, func(t *testing.T) {
 			var result string

--- a/z_heterogeneous_test.go
+++ b/z_heterogeneous_test.go
@@ -29,6 +29,7 @@ func TestHeterogeneousPoolIntegration(t *testing.T) {
 		t.Fatal(err)
 	}
 	cs.Heterogeneous = true
+	cs.EnableEvents = false
 	username := cs.Username
 	testHeterogeneousConStr := cs.StringWithPassword()
 	t.Log(testHeterogeneousConStr)

